web/src/main/java/org/springframework/security/web/csrf/CookieCsrfTokenRepository.java

```diff
@@ -63,7 +65,18 @@ public final class CookieCsrfTokenRepository implements CsrfTokenRepository {
 
 	private int cookieMaxAge = -1;
 
-	public CookieCsrfTokenRepository() {
+	@Nullable
+	private Consumer<ResponseCookie.ResponseCookieBuilder> cookieInitializer = null;
+
+	/**
+	 * Add a {@link Consumer} for a {@code ResponseCookieBuilder} that will be invoked
+	 * for each cookie being built, just before the call to {@code build()}.
+	 * @param initializer consumer for a cookie builder
+	 * @since 6.1
+	 */
+	public void addCookieInitializer(Consumer<ResponseCookie.ResponseCookieBuilder> initializer) {
```

Comment:
I don't think this property is initializing the cookie since it is already being initialized by the defaults in `saveToken`. Instead, it seems common in Spring Security to call this a customizer.

Also, instead of performing the `andThen` for the application, I think that the application should bring it's fully-composed component. That is, I think that `andThen` is too strong of an opinion for a general-purpose component.

Given these two, I'd recommend calling the method `setCookieCustomizer`.

Reply:
Yup, no problem. I implemented the change.